connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -104,11 +167,31 @@ public void onStateChanged(
 
         if (currentState == ConnectButtonState.Enabled) {
             setUpPolling();
+            if (permissionCallback != null && !permissionChecker.hasLocationPermission()) {
+                permissionCallback.onLocationPermissionRequested();
+            }
         }
     }
 
     @Override
     public void onError(ErrorResponse errorResponse) {
         // No-op
     }
+
+    interface LocationPermissionCallback {
+        void onLocationPermissionRequested();
+    }
+
+    private static final class LocationPermissionChecker {
```

Comment:
Why do we need this class? Looks like this is just a wrapper of the `ContextCompat.checkSelfPermission`, seems unnecessary to me.